### PR TITLE
GUI/menu/server/29

### PR DESCRIPTION
**Pull request by osvegn**
It implements a server menu at the left of the screen. This is a basic menu, without information inside for the moment. There is only interaction and scale adaptation with others menu.

### CODE REVIEW

File: src/zappy_gui_src/Core/SFML/Interface/ServerCommunication/ServerCommunication.hpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+            inline const sf::Vector2f &getPosition() const {
+                return _body.getPosition();
+            };
+            inline const sf::Vector2f &getSize() const {
+                return _body.getSize();
+            };
+            void update(bool forceUpdate = false);
+
+        private:
+            void _updateBody();
```
Comment (LucasTesnier) [inline src/zappy_gui_src/Core/SFML/Interface/ServerCommunication/ServerCommunication.hpp:35]: Missing documentation

**Review by LucasTesnier — CHANGES_REQUESTED** [review]
Please do an abastract class

**Review by GuyomT — APPROVED** [review]
Just add documentation where Lucas said otherwise LGTM

File: src/zappy_gui_src/Core/SFML/Interface/ServerCommunication/ServerCommunication.hpp (hunk 1 above)
Comment (osvegn) [reply src/zappy_gui_src/Core/SFML/Interface/ServerCommunication/ServerCommunication.hpp:35]: ups, I missed it

**Comment by osvegn** [comment]
> Please do an abastract class

an abstract for ?

**Review by LucasTesnier — APPROVED** [review]
NTR!

**Review by GuyomT — APPROVED** [review]
LGTM

**Pull request merged**